functionary/train/custom_datasets.py

```diff
@@ -518,7 +634,11 @@ def __getitem__(self, i) -> Dict[str, torch.Tensor]:
         if i in self.cached_data_dict:
             return self.cached_data_dict[i]
 
-        ret = prepare_training_inputs(self.raw_data[i], self.tokenizer)
+        ret = prepare_training_inputs(
```

Comment:
If we use: map_raw_data_to_input_dics it is not lazy any more, because map_raw_data_to_input_dics will tokenize all the data points first and have to wait. Actually this is: CustomDatasets without packing + not lazy

Reply:
Ok